packages/server/src/koaJsonRpc/index.ts

```diff
@@ -172,8 +173,24 @@ export default class KoaJsonRpc {
 
       ctx.body = jsonResp(body.id, null, result);
       if (result instanceof JsonRpcError) {
-        ctx.status = result.code == -32603 ? 500 : 400;
-        ctx.state.status = `${ctx.status} (${JSON_RPC_ERROR})`;
+        // What Status code to return for JsonRpcError
+        switch (result.code) {
```

Comment:
Q: Should we have additional test cases for status code or do we already have this covered?

Reply:
✅  Done.

I first tried to do it through UnitTest for `server.ts` but since the artifact only exports the KOA app, and there is no way to inject dependencies, there was no way for me to mock them, and the current UT for that class is only testing the errors and validations that occur prior to the mirror node calls, and excludes exceptions that might come from the mirror node or relay, since is not possible to mock them or even to spy on them to stub the responses.

That said, I created an acceptance test that instead of using the relayImpl directly to perform the `eth_call` call, it does so using an http testClient that gets the whole http response with its corresponding status code that we can assert.